client/state/utils.js

```diff
@@ -116,3 +194,45 @@ export function createReducer( initialState = null, customHandlers = {}, schema
 		return state;
 	};
 }
+
+/**
+ * Creates schema-validating reducer
+ *
+ * Use this to wrap simple reducers with a schema-based
+ * validation check when loading the initial state from
+ * persistent storage.
+ *
+ * When this wraps a reducer with a known JSON schema,
+ * it will intercept the DESERIALIZE action (on app boot)
+ * and check if the persisted state is still valid state.
+ * If so it will return the persisted state, otherwise
+ * it will return the initial state computed from
+ * passing the null action.
+ *
+ * @example
+ * const ageReducer = ( state = 0, action ) =>
+ *     GROW === action.type
+ *         ? state + 1
+ *         : state
+ *
+ * const schema = { type: 'number', minimum: 0 }
+ *
+ * export const age = withSchemaValidation( schema, age )
+ *
+ * // ageReducer( -5, { type: DESERIALIZE } ) === -5
+ * // age( -5, { type: DESERIALIZE } ) === 0
+ * // age( 23, { type: DESERIALIZE } ) === 23
```

Comment:
I added tests in 345da31 but I left the examples in the JSDoc for now. I'm open tot taking them out, but since these seem non-obvious looking at them I want to leave these here for those who read through this file but not the README.